### PR TITLE
Distribute prebuilt JupyterLab extension / Notebook Classic extension

### PR DESCRIPTION
**Pull request by timkpaine**

### CODE REVIEW

File: .husky/pre-push [hunk 1]
```diff
@@ -1,4 +1,4 @@
 #!/bin/sh
 . "$(dirname "$0")/_/husky.sh"
 
-yarn prepush
+#  yarn prepush
```
Comment (timkpaine) [inline .husky/pre-push:4]: remove before merge

File: azure-pipelines.yml [hunk 2]
```diff
@@ -183,6 +189,20 @@ parameters:
           codeCoverageTool: Cobertura
           summaryFileLocation: "$(System.DefaultWorkingDirectory)/**/*coverage.xml"
 
+  - name: pythonJSAssetsSteps
+    type: stepList
+    default:
+
+      - bash: |
+          node scripts/_pull_pipeline_artifact.js $(System.DefaultWorkingDirectory) $(Build.BuildId)
```
Comment (timkpaine) [inline azure-pipelines.yml:197]: add comments

File: python/perspective/perspective/extension/finos-perspective-jupyterlab.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+    "ServerApp": {
+      "jpserver_extensions": {
+        "perspective": true
```
Comment (timkpaine) [inline python/perspective/perspective/extension/finos-perspective-jupyterlab.json:4]: I don't think I need this one, double check

**Comment by timkpaine** [comment]
Confirmed working for JupyterLab
Screen Shot 2022-05-06 at 20 05 37

**Comment by timkpaine** [comment]
Closer for notebook/colab 
Screen Shot 2022-05-07 at 17 38 26

**Comment by timkpaine** [comment]
needs https://github.com/finos/perspective/pull/1801/files

**Comment by timkpaine** [comment]
Working now in both
Screen Shot 2022-05-07 at 11 18 08 PM

**Comment by timkpaine** [comment]
And Voilà with no CDN:
Screen Shot 2022-05-07 at 11 20 56 PM

**Comment by timkpaine** [comment]
Fixes:
https://github.com/finos/perspective/issues/1773

Should make these easier:
https://github.com/finos/perspective/issues/1251
https://github.com/finos/perspective/issues/942

File: python/perspective/package.json [hunk 4]
```diff
@@ -4,6 +4,11 @@
   "version": "1.3.11",
   "scripts": {
     "bench": "python3 bench/perspective_benchmark.py",
-    "docs": "python3 docs/generate.py"
+    "docs": "python3 docs/generate.py",
+    "clean:labextension": "rimraf perspective/labextension",
+    "clean:nbextension": "rimraf perspective/nbextension/index.js"
```
Comment (timkpaine) [inline python/perspective/package.json:9]: Remove whole static folder

File: python/perspective/package.json [hunk 5]
```diff
@@ -4,6 +4,11 @@
...
+  },
+  "devDependencies": {
+    "rimraf": "^3.0.2"
```
Comment (timkpaine) [inline python/perspective/package.json:12]: Move up to top level

**Comment by timkpaine** [comment]
this PR is now working across Jupyter, JupyterLab, and Voilà.

Screen Shot 2022-05-14 at 20 12 07

Screen Shot 2022-05-14 at 19 39 52
Screen Shot 2022-05-14 at 20 11 21


last step is to finish the python build process in azure to ensure the JS assets are packaged and distributed from the JS build to the python build

**Comment by timkpaine** [comment]
Wait for https://github.com/finos/perspective/pull/1808 which should already be setup to copy `nbextension` and `labextension` assets from the JS build to the python builds

**Comment by texodus** [comment]
@timkpaine This branch has become stale/unwieldy.  I've backed out the merge commits, squashed the remainder and rebased to master, cleaning up a few conflicts with the most in `setup.py`.  Result is in #1824, closing this PR in favor of that one.

**Comment by timkpaine** [comment]
Reopening due to bad rebase

**Comment by timkpaine** [comment]
I will rebase at the very end when I've cleaned stuff up and can make sure the diff is clean

**Comment by timkpaine** [comment]
moving to new branch

**Comment by Radouaneb** [comment]
I'm not able to load jupyterLab extension. is it normal ?

**Pull request closed without merging**